editor/editor_node.cpp

```diff
@@ -1253,19 +1272,18 @@ void EditorNode::_viewport_resized() {
 }
 
 void EditorNode::_titlebar_resized() {
+	if (!left_menu_spacer->is_visible()) {
+		// Not in expand to title mode.
+		return;
+	}
 	DisplayServer::get_singleton()->window_set_window_buttons_offset(Vector2i(title_bar->get_global_position().y + title_bar->get_size().y / 2, title_bar->get_global_position().y + title_bar->get_size().y / 2), DisplayServer::MAIN_WINDOW_ID);
 	const Vector3i &margin = DisplayServer::get_singleton()->window_get_safe_title_margins(DisplayServer::MAIN_WINDOW_ID);
-	if (left_menu_spacer) {
-		int w = (gui_base->is_layout_rtl()) ? margin.y : margin.x;
-		left_menu_spacer->set_custom_minimum_size(Size2(w, 0));
-	}
-	if (right_menu_spacer) {
-		int w = (gui_base->is_layout_rtl()) ? margin.x : margin.y;
-		right_menu_spacer->set_custom_minimum_size(Size2(w, 0));
-	}
-	if (title_bar) {
-		title_bar->set_custom_minimum_size(Size2(0, margin.z - title_bar->get_global_position().y));
-	}
+
+	int w = (gui_base->is_layout_rtl()) ? margin.y : margin.x;
+	left_menu_spacer->set_custom_minimum_size(Size2(w, 0));
+	right_menu_spacer->set_custom_minimum_size(Size2(w, 0));
```

Comment:
Spacers are there to reserve space for native window close/min/max buttons on macOS. It should use opposite X/Y values for different sides, not a single W for both.